sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero.cpp

```diff
@@ -1257,9 +1258,10 @@ bool CopyEngineRequested(const ur_device_handle_t &Device) {
 // The default is standard commandlists. Setting 1 or 2 specifies use of
 // immediate commandlists.
 
-// Get value of immediate commandlists env var setting or -1 if unset
+// Get value of immediate commandlists env var setting or -1 if unset.
+// Also return whether a default or explicit setting is being returned.
 _ur_device_handle_t::ImmCmdlistMode
-_ur_device_handle_t::useImmediateCommandLists() {
+_ur_device_handle_t::useImmediateCommandLists(bool &Default) {

```

Comment:
This interface becomes hard to read this way and often requires dummy (unused) input. I suggest just check `std::getenv("SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS")` directly at the only place that needs to know (temporarily) about explicit setting. Or, move `ImmediateCommandlistsSetting` to a global scope and use that.

Reply:
Ok, I've split out the env var check separately as a temporary measure.